### PR TITLE
Add more details to transaction details gui.

### PR DESCRIPTION
**Pull request by tomasbrod**
* Details from boincHash for coin stake transactios
* a little more info for standard transactions (type of message if any)
* move some declarations to headers

### CODE REVIEW

File: src/qt/transactiondesc.cpp [hunk 1]
```diff
@@ -108,9 +110,13 @@ QString TransactionDesc::toHTML(CWallet *wallet, CWalletTx &wtx)
     //
     // From
     //
-    if (wtx.IsCoinBase() || wtx.IsCoinStake())
+    if (wtx.IsCoinBase())
     {
-        strHTML += "<b>" + tr("Source") + ":</b> " + tr("Generated") + "<br>";
+        strHTML += "<b>" + tr("Source") + ":</b> " + tr("Generated in CoinBase") + "<br>";
+    }
+    else if (wtx.IsCoinBase() || wtx.IsCoinStake())
```
Comment (denravonska) [inline src/qt/transactiondesc.cpp]: Should `wtx.IsCoinBase()` be here as well?

File: src/rpcrawtransaction.cpp [hunk 2]
```diff
@@ -29,43 +29,106 @@ extern Upgrader upgrader;
...
+        res["xbbNeuralHash"]=bb.NeuralHash;
+        res["xbbCurrentNeuralHash"]=bb.CurrentNeuralHash;
+        res["xbbNeuralContractSize"]=bb.superblock.length();
     }
-    pindexPrev = (*mi).second;
-    if (!block.ReadFromDisk(pindexPrev))
+    else
     {
-            return ""; //failed to read
+        if(bb.superblock.length()>=20)
```
Comment (denravonska) [inline src/rpcrawtransaction.cpp:82]: You could change these to:
`res["IsSuperBlock"] = (block.nIsSuperBlock != 0);`

or:
`res["IsSuperBlock"] = IsSuperBlock(&block);`

Comment (tomasbrod) [reply src/rpcrawtransaction.cpp:82]: You mean pblockIndex?
I guess I can, pindex flags should be consistent with block, but i wanted to display the value from boinchash.

File: src/qt/transactiondesc.cpp (hunk 1 above)
Comment (tomasbrod) [reply src/qt/transactiondesc.cpp]: True, i will remove it.

File: src/rpcrawtransaction.cpp (hunk 2 above)
Comment (denravonska) [reply src/rpcrawtransaction.cpp:82]: Ah, I see. You know what, leave that for now. We can add a `IsValidSuperBlock(const std::string&)` and replace all the magic length checks later.

**Pull request merged**